feat: APPS-3147 Create FlexibleBlockInfo

tinuola opened the pull request:
Connected to [APPS-3147](https://jira.library.ucla.edu/browse/APPS-3147), [APPS-2785](https://uclalibrary.atlassian.net/browse/APPS-2785)

**Component Updated:** BlockInfo.vue, IconWithLink.vue

**Component Created:** FlexibleInfo.vue

**Stories:** 
- ~/stories/BlockInfo.stories.js
- ~/stories/Flexible_Info.stories.js

**Spec:** ~/stories/Flexible_Info.spec.js

**Notes:**
- Refactored `BlockInfo` to be usable for generic info display;
  - Removed FTVA specific props; added slots
  - __Note__: Tested the refactored component locally on Nuxt; see screenshots below
  - __Note__: Page-level updates will be needed; see ticket [APPS-3231](https://uclalibrary.atlassian.net/browse/APPS-3231)
- Created `FlexibleInfo` component
- Updated `IconWithLink` with new FTVA icons
- Updated design tokens to `5.30.0`

**Storybook Previews:**
- FTVA BlockInfo
  - _Original_: https://deploy-preview-692--ucla-library-storybook.netlify.app/?path=/story/block-info--default

  - _Refactored Default_: https://deploy-preview-694--ucla-library-storybook.netlify.app/?path=/story/block-info--default-ftva-info
  - _Refactored Implementation_: https://deploy-preview-694--ucla-library-storybook.netlify.app/?path=/story/block-info--example-ftva-info

- FlexibleInfo
  - `infoBlock` _typehandle_: https://deploy-preview-694--ucla-library-storybook.netlify.app/?path=/story/flexible-info--ftva-info-block
  - `contactInfoBlock` _typehandle_: https://deploy-preview-694--ucla-library-storybook.netlify.app/?path=/story/flexible-info--ftva-contact-info-block

<br>

**Local Dev - Components:**

![local-dev-preview](https://github.com/user-attachments/assets/0f69b8d1-0d74-4ce4-b4c1-f8901fa3f03a)

<br>

**Local Nuxt - Series page:**

![local-nutx-series-page](https://github.com/user-attachments/assets/334b4503-ca01-4e03-880a-7db27addc898)

<br>

**Local Nuxt - Event page:**

![local-nuxt-event-page](https://github.com/user-attachments/assets/ad28fd0d-ddbf-4417-b89a-3718a67588a4)


**Checklist:**

-   [x] I checked that it is working locally in the dev server
-   [x] I checked that it is working locally in the storybook
-   [x] I checked that it is working locally in the library-website-nuxt dev server
-   [x] I added a screenshot of it working
-   [x] UX has reviewed and approved this
-   [x] I have requested a PR review from the Dev team
-   [x] I used a conventional commit message
-   [x] I assigned myself to this PR


[APPS-3147]: https://uclalibrary.atlassian.net/browse/APPS-3147?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[APPS-2785]: https://uclalibrary.atlassian.net/browse/APPS-2785?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[APPS-3231]: https://uclalibrary.atlassian.net/browse/APPS-3231?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

farosFreed reviewed (dismissed):
This looks really good. I'm wondering if the 2 Col Layout stories should be refactored though? Or at least remove the old InfoBlock with the incorrect data if its too much code duplication

https://deploy-preview-692--ucla-library-storybook.netlify.app/?path=/story/layout-2-column-layout-with-sticky-sidebar--event-series
vs
https://deploy-preview-694--ucla-library-storybook.netlify.app/?path=/story/layout-2-column-layout-with-sticky-sidebar--event-series

tinuola commented:
@farosFreed Updated the 2ColLayout stories. I refactored BlockInfo in one story and removed it completely from another since it would be duplicating the same scenario/content. Thanks for the catch!

> This looks really good. I'm wondering if the 2 Col Layout stories should be refactored though? Or at least remove the old InfoBlock with the incorrect data if its too much code duplication
> 
> https://deploy-preview-692--ucla-library-storybook.netlify.app/?path=/story/layout-2-column-layout-with-sticky-sidebar--event-series vs https://deploy-preview-694--ucla-library-storybook.netlify.app/?path=/story/layout-2-column-layout-with-sticky-sidebar--event-series

pghorpade reviewed (approved):
👍

pghorpade commented:
:tada: This PR is included in version 3.56.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/ucla-library-website-components/v/3.56.0)
- [GitHub release](https://github.com/UCLALibrary/ucla-library-website-components/releases/tag/v3.56.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
